integrations/amazon_bedrock/tests/test_chat_generator.py

```diff
@@ -55,128 +52,242 @@ def tools():
+            streaming_callback=print_streaming_chunk,
+            tools=[tool],
+        )
 
-    layer = AmazonBedrockChatGenerator(
-        model="anthropic.claude-3-5-sonnet-20240620-v1:0", generation_kwargs=generation_kwargs
-    )
-    assert layer.generation_kwargs == generation_kwargs
+        pipeline = Pipeline()
+        pipeline.add_component("generator", generator)
 
+        pipeline_dict = pipeline.to_dict()
+        expected_dict = {
+            "metadata": {},
+            "max_runs_per_component": 100,
+            "connection_type_validation": True,
+            "components": {
+                "generator": {
+                    "type": CLASS_TYPE,
+                    "init_parameters": {
+                        "aws_access_key_id": {"type": "env_var", "env_vars": ["AWS_ACCESS_KEY_ID"], "strict": False},
+                        "aws_secret_access_key": {
+                            "type": "env_var",
+                            "env_vars": ["AWS_SECRET_ACCESS_KEY"],
+                            "strict": False,
+                        },
+                        "aws_session_token": {"type": "env_var", "env_vars": ["AWS_SESSION_TOKEN"], "strict": False},
+                        "aws_region_name": {"type": "env_var", "env_vars": ["AWS_DEFAULT_REGION"], "strict": False},
+                        "aws_profile_name": {"type": "env_var", "env_vars": ["AWS_PROFILE"], "strict": False},
+                        "model": "anthropic.claude-3-5-sonnet-20240620-v1:0",
+                        "generation_kwargs": {"temperature": 0.7},
+                        "stop_words": ["eviscerate"],
+                        "streaming_callback": "haystack.components.generators.utils.print_streaming_chunk",
+                        "boto3_config": None,
+                        "tools": [
+                            {
+                                "type": "haystack.tools.tool.Tool",
+                                "data": {
+                                    "name": "weather",
+                                    "description": "useful to determine the weather in a given location",
+                                    "parameters": {"city": {"type": "string"}},
+                                    "function": "tests.test_chat_generator.weather",
+                                },
+                            }
+                        ],
+                    },
+                }
+            },
+            "connections": [],
+        }
 
-def test_constructor_with_empty_model():
-    """
-    Test that the constructor raises an error when the model is empty
-    """
-    with pytest.raises(ValueError, match="cannot be None or empty string"):
-        AmazonBedrockChatGenerator(model="")
+        if not hasattr(pipeline, "_connection_type_validation"):
+            expected_dict.pop("connection_type_validation")
+
+        # add outputs_to_string, inputs_from_state and outputs_to_state tool parameters for compatibility with
```

Comment:
these can be added to the`expected_dict`. We always test with the latest version of Haystack